dpp/src/data_contract/enrich_data_contract_with_base_schema.rs

```diff
@@ -0,0 +1,63 @@
+use crate::{errors::ProtocolError, util::json_schema::JsonSchemaExt};
+
+use super::DataContract;
+use anyhow::anyhow;
+use serde_json::Value as JsonValue;
+
+const PROPERTY_PROPERTIES: &str = "properties";
+const PROPERTY_REQUIRED: &str = "required";
+
+pub fn enrich_data_contract_with_base_schema(
+    data_contract: &DataContract,
+    base_schema: JsonValue,
+    schema_id_byte_prefix: u8,
+    exclude_properties: &[&str],
+) -> Result<DataContract, ProtocolError> {
+    let mut cloned_data_contract = data_contract.clone();
+    cloned_data_contract.schema = String::from("");
+
+    let base_properties = base_schema
+        .get_schema_properties()?
```

Comment:
Does this actually work for `JsonValue`? Or is there an additional trait on json value that provides this?